### PR TITLE
Longopts support for dimfilter and grdfilter.

### LINKED ISSUE #7595: Longoptions support for dimfilter and grdfilter
**Description of the desired feature**

Longoptions support for dimfilter and grdfilter

**Are you willing to help implement and maintain this feature?** Yes

### PR DESCRIPTION
**Pull request by rbdavis**
**Description of proposed changes**

Longopts support for dimfilter and grdfilter.

Fixes #7595

**Reminders**

- [ ] Make sure that your code follows our style. Use the other functions/files as a basis.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Describe changes to function behavior and arguments in a comment below the function declaration.
- [ ] If adding new functionality, add a detailed description to the documentation and/or an example.

### CODE REVIEW

**Comment by joa-quim** [comment]
I must call the attention that both GMT.jl and PyGMT are using different option names in ``grdfilter``. That is something we want avoid, or at least minimize.

File: src/longopt/grdfilter_inc.h [hunk 1]
```diff
@@ -24,27 +24,19 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = {
...
-	          "",                  "" },
-	{ 0, 'I', "",
-	          "",                  "",
-	          "",                  "" },
-	{ 0, 'N', "",
-	          "",                  "",
-	          "",                  "" },
-	{ 0, 'T', "",
-	          "",                  "",
+	{ 0, 'D', "disttype",     "", "", "", "" },
```
Comment (joa-quim) [inline src/longopt/grdfilter_inc.h:27]: Both GMT.jl and PyGMT use "distance". At least add this as alias

File: src/longopt/grdfilter_inc.h [hunk 2]
```diff
@@ -24,27 +24,19 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = {
...
+	{ 0, 'F', "filter",
+	          "b,c,g,f,o,m,p,h,l,L,u,U",
+ 				       "boxcar,cosarch,gaussian,custom,operator,median,mlprob,histogram,minall,minpos,maxall,maxneg",
+	          "c,h,l,q,u",         "center,highpass,lower,quantile,upper" },
+	{ 0, 'G', "outgrid",
+	          "",             "",
+	          "d,n,o,s,c",         "divide,nan,offset,scale,gdal" },
+	GMT_INCREMENT_KW,              /* quasi-common -I option from gmt_constants.h */
+	{ 0, 'N', "nan",
```
Comment (joa-quim) [inline src/longopt/grdfilter_inc.h:36]: GMT.jl & PyGMT use "nans"

File: src/longopt/grdfilter_inc.h [hunk 3]
```diff
@@ -24,27 +24,19 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = {
...
+	          "i,p,r",             "ignore,coregnan,anynan",
 	          "",                  "" },
+	{ 0, 'T', "togglereg",         "", "", "", "" },
```
Comment (joa-quim) [inline src/longopt/grdfilter_inc.h:39]: Again,the wrappers use "toggle"

File: src/longopt/dimfilter_inc.h [hunk 4]
```diff
@@ -24,39 +24,20 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = {
...
-	          "",                  "",
-	          "",                  "" },
-	{ 0, 'S', "",
-	          "",                  "",
-	          "",                  "" },
-	{ 0, 'T', "",
-	          "",                  "",
-	          "",                  "" },
+	{ 0, 'D', "disttype",     "", "", "", "" },
+	{ 0, 'F', "prifilter",
```
Comment (joa-quim) [inline src/longopt/dimfilter_inc.h:28]: I don't like "prifilter" very much. My proposal is to call it just "filter" and let the -N filter be called "filter2" or "secondary_filter" (too long?).

Comment (PaulWessel) [reply src/longopt/dimfilter_inc.h:28]: OK, filter sounds good to me and then let the other be secondary_filter, maybe with alias secfilter

**Comment by PaulWessel** [comment]
For other comments and suggestions please post them by end of this work week (Friday)

**Comment by PaulWessel** [comment]
Thanks @joa-quim for comments, I agree:

grdfilter:

-D is better with --distance
-N is better with --nans
-T:Let's do --toggle_registration with alias --toogle
-F: --filter is fine

dimfilter:
-D is better with --distance
-F: --filter is fine
-N we do --secondary_filter with alias secfilter

Last call to other members of @GenericMappingTools/long-options-review. By Saturday I will pass baton to @roger to make changes to branch then approve and merge.

**Review by PaulWessel — APPROVED** [review]
Good to go

File: src/longopt/grdfilter_inc.h (hunk 1 above)
Comment (PaulWessel) [reply src/longopt/grdfilter_inc.h]: Done

File: src/longopt/grdfilter_inc.h (hunk 2 above)
Comment (PaulWessel) [reply src/longopt/grdfilter_inc.h]: Done

File: src/longopt/grdfilter_inc.h (hunk 3 above)
Comment (PaulWessel) [reply src/longopt/grdfilter_inc.h]: Done

File: src/longopt/dimfilter_inc.h (hunk 4 above)
Comment (PaulWessel) [reply src/longopt/dimfilter_inc.h]: Done

File: src/longopt/dimfilter_inc.h [hunk 5]
```diff
@@ -24,39 +24,20 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = {
...
-	          "",                  "" },
+	{ 0, 'D', "distance",     "", "", "", "" },
+	{ 0, 'F', "filter",
+	          "b,c,g,m,p",    "boxcar,cosarch,gaussian,median,maxprob",
+	          "l,u",          "lower,upper" },
+	{ 0, 'G', "outgrid",
+	          "",             "",
+	          "d,n,o,s",      "divide,nan,offset,scale" },
+	GMT_INCREMENT_KW,         /* quasi-common -I option from gmt_constants.h */
+	{ 0, 'L', "stdoutscript", "", "", "", "" },
```
Comment (joa-quim) [inline src/longopt/dimfilter_inc.h:35]: Most people don't know what is ``stdout``. Better if we call it ``bashscript``, which is what it does.

Comment (PaulWessel) [reply src/longopt/dimfilter_inc.h:35]: Agree missed that one

Comment (PaulWessel) [reply src/longopt/dimfilter_inc.h]: Chose `--script `since may not be bash only (even though it is now).

**Review by seisman — APPROVED** [review]

File: src/longopt/dimfilter_inc.h [hunk 6]
```diff
@@ -24,39 +24,20 @@ static struct GMT_KEYWORD_DICTIONARY module_kw[] = {
...
+	{ 0, 'D', "distance",     "", "", "", "" },
+	{ 0, 'F', "filter",
+	          "b,c,g,m,p",    "boxcar,cosarch,gaussian,median,maxprob",
+	          "l,u",          "lower,upper" },
+	{ 0, 'G', "outgrid",
+	          "",             "",
+	          "d,n,o,s",      "divide,nan,offset,scale" },
+	GMT_INCREMENT_KW,         /* quasi-common -I option from gmt_constants.h */
+	{ 0, 'L', "script", "", "", "", "" },
+	{ 0, 'N', "secondary_filter|secfilter",
```
Comment (seisman) [inline src/longopt/dimfilter_inc.h:36]: PyGMT uses `sectors` for `-N`, but it must be a mistake.

Comment (PaulWessel) [reply src/longopt/dimfilter_inc.h]: Not really. Turns out there is a sector setting as part of the filter setting and it does a filtering of the results across the sectors.  I renamed it sector_filter with alias secfilter

**Pull request merged**